sdk/nodejs/cmd/run-policy-pack/run.ts

```diff
@@ -0,0 +1,264 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as fs from "fs";
+import * as minimist from "minimist";
+import * as path from "path";
+import * as tsnode from "ts-node";
+import { ResourceError, RunError } from "../../errors";
+import * as log from "../../log";
+import { disconnectSync } from "../../runtime/settings";
+import { runInPulumiStack } from "../../runtime/stack";
+
+// Keep track if we already logged the information about an unhandled error to the user..  If
+// so, we end with a different exit code.  The language host recognizes this and will not print
+// any further messages to the user since we already took care of it.
+//
+// 32 was picked so as to be very unlikely to collide with any of the error codes documented by
+// nodejs here:
+// https://github.com/nodejs/node-v0.x-archive/blob/master/doc/api/process.markdown#exit-codes
+export const nodeJSProcessExitedAfterLoggingUserActionableMessage = 32;
```

Comment:
Seems we copy/pasted but even undid some sharing that existed in the source (where this was shared between index and run).  Best case we share all of this across both plugin and normal langhost execution, but if we do need to copy some, let's keep at least the existing level of sharing.

Reply:
Discussed offline, I'm going to mark this and the above questions about the copy/paste here resolved. The reasons for doing this are mentioned in the PR comment, so if you're following along, see there for the reasoning.